CONTRIBUTING.md

```diff
@@ -0,0 +1,58 @@
+# Contributing
+
+## Folder Structure
+
+The basic idea here is that we want to balance between flexibility and stability while ensuring anything we do produce is easily understood, used and modified by another person. We basically do this by having a core library, `swarm` that can be relied upon and an `experiments` dir that builds on that base in any direction it feels.
+
+1. `swarm` contains all the core code that we rely upon - any changes here should be done via a Merge Request
+    - `tests` are checks for `swarm`'s capabilities and should be treated as an extension of `swarm`
+    - conceptually any code shared across experiments should go into swarm
+
+2. `experiments` is where the research goes - this is where we can test our ideas in self-contained folders and generate animations + writeups on the results. Experiments is used as a way to share code/ideas rather than to produce libraries and reusable code- anything good should be merged into swarm and a deletion of an experiment should have no flow on effects.
+
+3. `sample` is a place to put animations/notebooks that give a sense of what we're doing here. It's a non critical folder, but there is value in having some sort of minimal examples to share instantly.
+    - There is a potential to run a folder like this to produce the animations/writeups in a nice format.
+
+## Suggested Git Workflow
+
+General Philosophies:
+
+1. Many small changes are easier than few big changes. 
+2. Notebooks are not reviewable and it is discouraged as a way of collaborating. They work as finished products and places to try ideas quickly, rather than something that can be shared and worked collaboratively.
+    - possibly use Colaboratory to work on notebooks together (do they have google docs workflows)
+
```

Comment:
Ok cool, happy to put changes worth reviewing in a function or something in a module.